### PR TITLE
feat(insights): InstantSearch Insights Integration

### PR DESCRIPTION
**Pull request by Aallam**
Integration of InstantSearch Insights:

- [x] Move and transform legacy Insights library
- [x] Support Insights + InstantSearch Widgets

### CODE REVIEW

File: helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/FilterTracker.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+     */
+    val trackerScope: CoroutineScope
+
+    /**
+     * Track a filter click event.
+     *
+     * @param filter filter to track
+     * @param customEventName custom event name, overrides the default event name
+     */
+    public fun <F : Filter> trackClick(filter: F, customEventName: String? = null)
```
Comment (spinach) [inline helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/FilterTracker.kt:32]: I would say that we change `customEventName` to `eventName`. In iOS, we have `eventName customEventName: String? = nil` which means the label to the developer is `eventName` but the inner local variable used is `customEventName`

Comment (spinach) [reply helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/FilterTracker.kt:32]: Also it keeps consistency with `eventName` that is used in other places

File: helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/HitsTracker.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    trackerScope = coroutineScope
+)
+
+/**
+ * Creates a [HitsTracker] object.
+ *
+ * @param eventName default event name
+ * @param searcher multiple index searcher
+ * @param pointer pointer to a specific index position
+ * @param insights actual events handler
```
Comment (spinach) [inline helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/HitsTracker.kt:75]: nitpicky: missing coroutineScope doc

File: helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/FilterTracker.kt (hunk 1 above)
Comment (Aallam) [reply helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/FilterTracker.kt]: Totally agree!
Fixed :D

File: helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/HitsTracker.kt (hunk 2 above)
Comment (Aallam) [reply helper/src/commonMain/kotlin/com/algolia/instantsearch/helper/tracker/HitsTracker.kt:76]: Nice catch! Fixed too

**Pull request merged**